addons/web/static/src/views/list/list_renderer.js

```diff
@@ -94,6 +94,7 @@ export class ListRenderer extends Component {
             this.state.columns = this.allColumns.filter(
                 (col) => !col.optional || this.optionalActiveFields[col.name]
             );
+            this.editableDirection = !nextProps.readonly ? nextProps.archInfo.editable : false;
```

Comment:
this is louche to only set this in willUpdateProps

Reply:
Suggestion: have a `editable` getter which returns `!this.props.readonly ? this.props.archInfo.editable : false`

Reply:
@aab-odoo I can use a getter, but why is it louche ? Since it's based on a prop, putting it there makes sure it's always with the correct value isn't it ? It's the same as the getter, just more ""optimized"

Reply:
because you **only** set it in willUpdateProps, so it's not set initially. Anyway, it's a good candidate for a getter imo

Reply:
I thought willUpdateProps was called once on init...